### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by foxtran**
I have cherry-picked the commits which can be safely merged with devel branch. 

The rest commits are still in #223

### CODE REVIEW

File: src/converters/converters.h [hunk 1]
```diff
@@ -95,19 +95,19 @@ template<class T>
 struct std_item
 {
     typedef typename T::value_type V;
-    static V& get(T const& x, int i)
+    static V& get(T & x, int i)
```
Comment (alexvakimov) [inline src/converters/converters.h:98]: for the changes in this file - i'm not sure if it is a good idea to omit the "const" keyword; i followed the example in the URL above - so that's basically what they do;  I think using non-const references could give more flexibility, but also opens more room for potential problems, so it would be less safe to do them; Please revert the changes in this file.

File: src/dyn/libdyn.cpp [hunk 2]
```diff
@@ -375,7 +375,7 @@ void export_dyn_decoherence_objects(){
   int instantaneous_decoherence_variant, int collapse_option) = &instantaneous_decoherence;
   def("instantaneous_decoherence", expt_instantaneous_decoherence_v1);
 
-  void (*expt_instantaneous_decoherence_dia_v1)(CMATRIX& Coeff, nHamiltonian& ham,
+  void (*expt_instantaneous_decoherence_dia_v1)(CMATRIX& Coeff,
```
Comment (alexvakimov) [inline src/dyn/libdyn.cpp:378]: even though the ham was not used in this function, it is there for a reason - since the function is only experimental, there maybe need to pass some Hamiltonian-related information to this function, so I did it only as a placeholder; also, if you change this signature here, one would need to also adjust the corresponding interface and implementation in the .h and .cpp files.

File: src/dyn/libdyn.cpp [hunk 3]
```diff
@@ -474,7 +474,7 @@ void export_dyn_decoherence_objects(){
   def("schwartz_1", expt_schwartz_1_v2);
 
   vector<MATRIX> (*expt_schwartz_2_v1)
-  (dyn_control_params& prms, CMATRIX& amplitudes, nHamiltonian& ham, MATRIX& inv_alp) = &schwartz_2;
+  (dyn_control_params& prms, nHamiltonian& ham, MATRIX& inv_alp) = &schwartz_2;
```
Comment (alexvakimov) [inline src/dyn/libdyn.cpp:477]: this is a good correction

File: src/libint2_wrappers/libint2_wrappers.cpp [hunk 4]
```diff
@@ -102,8 +102,6 @@ void add_to_shell(std::vector<libint2::Shell>& shells,
                        {{coords.x, coords.y, coords.z}}   // origin coordinates
                      }
                     );
-    // Return the shell with new data
-    return shells;
```
Comment (alexvakimov) [inline src/libint2_wrappers/libint2_wrappers.cpp:106]: agree

File: src/math_linalg/MATRIX.cpp [hunk 5]
```diff
@@ -102,23 +102,23 @@ the corresponding given input vectors
 
 
 
-MATRIX MATRIX::T(){   
+MATRIX MATRIX::T() const {
```
Comment (alexvakimov) [inline src/math_linalg/MATRIX.cpp:105]: agree for the changes in this file and the .h one

File: src/math_linalg/MATRIX3x3.h [hunk 6]
```diff
@@ -129,7 +129,7 @@ class MATRIX3x3{
     tmp = xz; xz = zx; zx = tmp;
     tmp = yz; yz = zy; yz = tmp;
   }
-  MATRIX3x3 T(){
+  MATRIX3x3 T()const {
```
Comment (alexvakimov) [inline src/math_linalg/MATRIX3x3.h:132]: agree

File: src/math_linalg/base_matrix.h [hunk 7]
```diff
@@ -172,10 +172,10 @@ class base_matrix{
   }
   
   ///< Returns the matrix element with the index "indx"
-  T1 get(int i){ return M[i]; }
+  T1 get(int i) const { return M[i]; }
```
Comment (alexvakimov) [inline src/math_linalg/base_matrix.h:175]: same here...

File: src/models/Model_DAC.cpp [hunk 8]
```diff
@@ -27,7 +27,7 @@ namespace libmodels{
 
 
 void model_DAC(CMATRIX& Hdia, CMATRIX& Sdia, vector<CMATRIX>& d1ham_dia, vector<CMATRIX>& dc1_dia,
-               vector<double> q, vector<double>& params){ 
+               vector<double> &q, vector<double>& params){
```
Comment (alexvakimov) [inline src/models/Model_DAC.cpp:30]: yes - this would be a better way

File: src/models/Model_DAC.h [hunk 9]
```diff
@@ -29,7 +29,7 @@ namespace libmodels{
 
 
 void model_DAC(CMATRIX& Hdia, CMATRIX& Sdia, vector<CMATRIX>& d1ham_dia, vector<CMATRIX>& dc1_dia,
-               vector<double> q, vector<double>& params);
+               vector<double> &q, vector<double>& params);
```
Comment (alexvakimov) [inline src/models/Model_DAC.h:32]: yes

**Review by alexvakimov — CHANGES_REQUESTED** [review]
Hi @foxtran - please revert a few changes; other changes would be fine - then i accept it

File: src/converters/converters.h (hunk 1 above)
Comment (foxtran) [reply src/converters/converters.h:98]: Ok. Let me think how to update it in a better way
